src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -81,29 +82,39 @@ public RuntimeConfig GetConfig()
                 subStatusCode: DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
         }
 
-        CheckForAndSetupConfigFileWatcher();
+        TrySetupConfigFileWatcher();
         return _runtimeConfig;
     }
 
     /// <summary>
     /// Checks if we are in a local development scenario, and if so, instantiate
     /// the config file watcher with a reference to this RuntimeConfigProvider. Otherwise
     /// we will call the no argument constructor, which means no file will actually be
-    /// watched.
+    /// watched. Returns true when setting up a non null config file watcher and false otherwise.
     /// </summary>
-    private void CheckForAndSetupConfigFileWatcher()
+    private bool TrySetupConfigFileWatcher()
     {
         if (_configFileWatcher is null)
         {
             if (!IsLateConfigured && _runtimeConfig is not null && _runtimeConfig.IsDevelopmentMode())
             {
-                _configFileWatcher = new(this);
+                FileSystemRuntimeConfigLoader loader = (FileSystemRuntimeConfigLoader)ConfigLoader;
+                string configFilePath = loader.ConfigFilePath;
+                string configFileName = Path.GetFileName(configFilePath);
+                string? directoryName = Path.GetDirectoryName(configFilePath);
+                directoryName = string.IsNullOrWhiteSpace(directoryName) ?
+                    loader.GetCurrentDirectoryFromFileSystem() :
+                    directoryName;
+                _configFileWatcher = new(this, directoryName, configFileName);
+                return true;
```

Comment:
Return true would be valid for both conditions where `_configFileWatcher = new(...)` is called, wouldn't it?